.github/workflows/docs.yml

```diff
@@ -0,0 +1,42 @@
+name: Auto-Generate Docs
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  generate-javadoc:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+
+    - name: Set up JDK 20
+      uses: actions/setup-java@v3
+      with:
+        java-version: '20'
+
+    - name: Create docs directory if it doesn't exist
+      run: mkdir -p ./docs
+
+    - name: Generate Javadoc
+      run: mvn javadoc:javadoc       
+
+    - name: Switch to docs-deployment branch
+      run: |
+        git fetch origin
+        git switch --create docs-deployment || git switch docs-deployment
+
+    - name: Copy Javadoc to docs-deployment branch
+      run: |
+        rm -rf *
+        cp -r ../docs/* ./
```

Comment:
Why should the docs lay a level above the working directory? Isn't the docs folder at ./docs?

Reply:
Changed it with new commit to behaviour described above